compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1765,7 +1765,20 @@ impl<'a> Parser<'a> {
     /// Parses an `if` expression (`if` token already eaten).
     fn parse_if_expr(&mut self, attrs: AttrVec) -> PResult<'a, P<Expr>> {
         let lo = self.prev_token.span;
-        let cond = self.parse_cond_expr()?;
+        let mut cond = self.parse_cond_expr()?.into_inner();
+
+        if let ExprKind::Paren(paren) = &cond.kind {
```

Comment:
We still need the feature gate check we talked about here, right?

Reply:
That's blocked currently. Have you seen my comment above? https://github.com/rust-lang/rust/pull/83407#discussion_r602687796

Reply:
Hadn't, looking into it. (The parser is indeed not supposed to be able to use that to avoid diverging parsing based on feature gates.)